### PR TITLE
Add some more statistics, Modify game ticks

### PR DESCRIPTION
**Pull request by RedSparr0w**
Add new statistics:
- Daily Deal Trades
- Total Berries Harvested
- Quests Completed
- Total Shards Gained

Fix game tick speed:
Should be checking if the current tick matches or above instead of just above, otherwise the tick doesn't happen until 1.1 seconds instead of 1 second.

### CODE REVIEW

File: src/scripts/Game.ts [hunk 1]
```diff
@@ -179,14 +181,19 @@ class Game {
...
+        EffectEngineRunner.counter += GameConstants.TICK_TIME;
+        if (EffectEngineRunner.counter >= GameConstants.EFFECT_ENGINE_TICK) {
             EffectEngineRunner.tick();
         }
 
-        if (GameHelper.counter > 60 * 1000) {
-            GameHelper.updateTime();
+        // Game timers
+        GameHelper.counter += GameConstants.TICK_TIME;
+        if (GameHelper.counter >= GameConstants.SECOND) {
```
Comment (Aegyo) [inline src/scripts/Game.ts]: Should this be `GameConstants.MINUTE`?

**Review by Aegyo — COMMENTED** [review]
:+1:
The version check may need an update after develop is merged up into this.
Also a potential addition to this - routeKills needs to be split into region and route before we add Unova, since those routes start back at 1. May prefer to do that in a separate PR though.

File: src/scripts/Game.ts (hunk 1 above)
Comment (RedSparr0w) [reply src/scripts/Game.ts]:
```suggestion
        if (GameHelper.counter >= GameConstants.MINUTE) {
```

**Comment by RedSparr0w** [comment]
> Also a potential addition to this - routeKills needs to be split into region and route before we add Unova, since those routes start back at 1. May prefer to do that in a separate PR though.

Will do that in a different PR, not entirely sure how we will go about it.
As it won't work with any of the current implementations,
Unless we use the object and merge the region and route number into 1 (`region_number|route_number`),
But that feels a little bit messy.

**Comment by RedSparr0w** [comment]
I've also added 2 more statistics since the last review:
- Pokemon Hatched
- Shiny Pokemon Hatched

I've added to the update function to pull our current method of hatched statistics into the new one.

**Review by Aegyo — APPROVED** [review]

**Pull request merged**